Riot/Modules/Room/RoomInfo/RoomInfoList/RoomInfoListViewController.swift

```diff
@@ -0,0 +1,407 @@
+            cell.textView.text = row.text
+            cell.textView.isEditable = false
+            cell.textView.isScrollEnabled = false
+            cell.textView.backgroundColor = .clear
+            cell.selectionStyle = .none
+            cell.contentView.backgroundColor = theme.headerBackgroundColor
+            cell.update(theme: theme)
+            cell.textView.textColor = theme.textSecondaryColor
+            cell.hideSectionSeparators = true
+            
+            return cell
+        }
+    }
+    
+}
+
+// MARK: - UITableViewDelegate
+
+extension RoomInfoListViewController: UITableViewDelegate {
+    
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        cell.backgroundColor = theme.backgroundColor
+        cell.selectedBackgroundView = UIView()
+        cell.selectedBackgroundView?.backgroundColor = theme.selectedBackgroundColor
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return sections[section].header
+    }
+    
+    func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        return sections[section].footer
+    }
+    
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let header = sections[section].header else {
+            return nil
+        }
+
+        let view: TableViewHeaderFooterView? = tableView.dequeueReusableHeaderFooterView()
+
+        view?.textView.text = header
+        view?.textView.font = .systemFont(ofSize: 13)
+        view?.textViewInsets = UIEdgeInsets(top: 16, left: 16, bottom: 8, right: 16)
+        view?.update(theme: theme)
+
+        return view
+    }
+
+    func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        guard let footer = sections[section].footer else {
+            return nil
+        }
+
+        let view: TableViewHeaderFooterView? = tableView.dequeueReusableHeaderFooterView()
+
+        view?.textView.text = footer
+        view?.textView.font = .systemFont(ofSize: 13)
+        view?.update(theme: theme)
+
+        return view
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        
+        let row = sections[indexPath.section].rows[indexPath.row]
+        row.action?()
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        let row = sections[indexPath.section].rows[indexPath.row]
+        
+        switch row.type {
+        default:
+            return UITableView.automaticDimension
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        if sections[section].header == nil {
+            return 8
```

Comment:
Prefer to use constants instead of magic numbers: `static let tableViewSectionInset: CGFloat = 8.0`

Reply:
Updated